WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -582,9 +585,32 @@ private static async Task WaitAndHandleResultOfTasksAsync(string logPrefix, para
 		}
 	}
 
-	public void WalletEnteredSendWorkflow(string walletName) => WalletsInSendWorkflow.TryAdd(walletName, 0);
+	public void WalletEnteredSendWorkflow(string walletName) => WalletsInSendWorkflow.TryAdd(walletName, false);
 
-	public void WalletLeftSendWorkflow(string walletName) => WalletsInSendWorkflow.Remove(walletName, out _);
+	public void WalletLeftSendWorkflow(Wallet wallet)
+	{
+		if (CoinJoinClientStates.TryGetValue(wallet.WalletName, out var stateHolder) && WalletsInSendWorkflow.TryRemove(wallet.WalletName, out bool needRestart))
+		{
+			if (needRestart)
+			{
+				Task.Run(async () => await StartAsync(wallet, stateHolder.StopWhenAllMixed, stateHolder.OverridePlebStop, CancellationToken.None).ConfigureAwait(false));
+			}
+		}
+	}
+
+	public void WalletEnteredTxPreview(Wallet wallet)
```

Comment:
I would name this functionally, instead of basing it on the UI names. Like `WalletEnteredSending`.